cms/djangoapps/contentstore/views/tests/test_block.py

```diff
@@ -3473,3 +3496,111 @@ def test_self_paced_item_visibility_state(self, store_type):
         # Check that in self paced course content has live state now
         xblock_info = self._get_xblock_info(chapter.location)
         self._verify_visibility_state(xblock_info, VisibilityState.live)
+
+
+@patch('xmodule.modulestore.split_mongo.caching_descriptor_system.CachingDescriptorSystem.applicable_aside_types',
+       lambda self, block: ['test_aside'])
+class TestUpdateFromSource(ModuleStoreTestCase):
+    """
+    Test update_from_source.
+    """
+
+    def setUp(self):
+        """
+        Set up the runtime for tests.
+        """
+        super().setUp()
+        key_store = DictKeyValueStore()
+        field_data = KvsFieldData(key_store)
+        self.runtime = TestRuntime(services={'field-data': field_data})
```

Comment:
Do you need to set up a `TestRuntime` here, with its own keystore and everything? I feel like it's better to just use the normal modulestore, which `ModuleStoreTestCase` sets up for you - then it's essentially the same in the test environment as it is in prod.

Reply:
@bradenmacdonald I'm not sure how to use that with the tools I have available for testing asides. This is based on how asides were tested elsewhere in this module, where I borrowed it from. The AsideTest class is based on XBlockAside, which requires a generated def_id, usage_id a runtime argument, apparently from which the def_id and usage_id are generated. I'm not fully certain what all of this is, just that it's required.

If there's some better way to do it I'm happy to, but if that's the case it would be better to change it elsewhere in these tests, too.

Reply:
Asides are a half-baked feature and I think only used by one group at MIT for one very specific thing? I don't think it's necessary to test them here. We can just say that this particular functionality doesn't support asides.